youtube_dl/extractor/moviefap.py

```diff
@@ -0,0 +1,123 @@
+            'description': 'Experienced MILF giving an Amazing Handjob',
+            'thumbnail': 'http://img.moviefap.com/a16:9w990r/thumbs/be/322032-20l.jpg',
+            'uploader_id': 'darvinfred06',
+            'display_id': 'experienced-milf-amazing-handjob',
+            'categories': ['Amateur', 'Masturbation', 'Mature', 'Flashing']
+        }
+    }, {
+        # quirky single-format case where the extension is given as fid, but the video is really an flv
+        'url': 'http://www.moviefap.com/videos/e5da0d3edce5404418f5/jeune-couple-russe.html',
+        'md5': 'fa56683e291fc80635907168a743c9ad',
+        'info_dict': {
+            'id': 'e5da0d3edce5404418f5',
+            'ext': 'flv',
+            'title': 'Jeune Couple Russe',
+            'description': 'Amateur',
+            'thumbnail': 'http://pic.moviefap.com/thumbs/e5/949-18l.jpg',
+            'uploader_id': 'whiskeyjar',
+            'display_id': 'jeune-couple-russe',
+            'categories': ['Amateur', 'Teen']
+        }
+    }]
+
+    @staticmethod
+    def __get_thumbnail_data(xml):
+
+        """
+        Constructs a list of video thumbnails from timeline preview images.
+        :param xml: the information XML document to parse
+        """
+
+        timeline = xml.find('timeline')
+        if timeline is None:
+            # not all videos have the data - ah well
+            return []
+
+        # get the required information from the XML
+        width = str_to_int(timeline.find('imageWidth').text)
+        height = str_to_int(timeline.find('imageHeight').text)
+        first = str_to_int(timeline.find('imageFirst').text)
+        last = str_to_int(timeline.find('imageLast').text)
+        pattern = timeline.find('imagePattern').text
+
+        # generate the list of thumbnail information dicts
+        thumbnails = []
+        for i in range(first, last + 1):
+            thumbnails.append({
+                'url': pattern.replace('#', str(i)),
+                'width': width,
+                'height': height
+            })
+        return thumbnails
+
+    def _real_extract(self, url):
+
+        # find the video ID
+        video_id = self._match_id(url)
+
+        # retrieve the page HTML
+        webpage = self._download_webpage(url, video_id)
+
+        # find the URL of the XML document detailing video download URLs
+        info_url = self._html_search_regex(r'flashvars\.config = escape\("(.+?)"', webpage, 'player parameters')
+
+        # download that XML
+        xml = self._download_xml(info_url, video_id)
+
+        # create dictionary of properties we know so far, or can find easily
+        info = {
+            'id': video_id,
+            'title': self._html_search_regex(r'<div id="view_title"><h1>(.*?)</h1>', webpage, 'title'),
+            'display_id': re.compile(self._VALID_URL).match(url).group('name'),
+            'thumbnails': self.__get_thumbnail_data(xml),
+            'thumbnail': xml.find('startThumb').text,
```

Comment:
No, default is ok. I've just emphasized it should not be fatal in order not to break extraction.